csrc/ir/base_nodes.cpp

```diff
@@ -383,6 +385,11 @@ Expr* Expr::withWritePredicate(kir::Predicate* predicate) {
   return result;
 }
 
+Expr* Expr::withOutputPreallocated() {
+  output_is_preallocated_ = true;
```

Comment:
Check the Expr is in a HostIrContainer and is of certain types that supports preallocated outputs.